capella2polarion/__main__.py

```diff
@@ -239,5 +238,122 @@ def render_documents(
         polarion_worker.update_documents(project_data.updated_docs, project)
 
 
+@cli.command()
+@click.option(
+    "--plugin-config-file",
+    type=click.File(mode="r", encoding="utf8"),
+    required=True,
+    envvar="CAPELLA2POLARION_SYNCHRONIZE_CONFIG",
+)
+@click.option(
+    "--document-rendering-config",
+    type=click.File(mode="r", encoding="utf8"),
+    envvar="CAPELLA2POLARION_DOCUMENT_CONFIG",
+)
+@click.option(
+    "--overwrite-layouts",
+    is_flag=True,
+    default=False,
+    envvar="CAPELLA2POLARION_OVERWRITE_LAYOUTS",
+)
+@click.option(
+    "--overwrite-numbering",
+    is_flag=True,
+    default=False,
+    envvar="CAPELLA2POLARION_OVERWRITE_NUMBERING",
+)
+@click.option(
+    "--synchronize-config",
+    type=click.File(mode="r", encoding="utf8"),
+    envvar="CAPELLA2POLARION_SYNCHRONIZE_CONFIG",
+)
+@click.option(
+    "--force-update",
+    is_flag=True,
+    envvar="CAPELLA2POLARION_FORCE_UPDATE",
+    default=False,
+)
+@click.option(
+    "--type-prefix",
+    type=str,
+    envvar="CAPELLA2POLARION_TYPE_PREFIX",
+    default="",
+)
+@click.option(
+    "--role-prefix",
+    type=str,
+    envvar="CAPELLA2POLARION_ROLE_PREFIX",
+    default="",
+)
+@click.option(
+    "--grouped-links-custom-fields / --no-grouped-links-custom-fields",
+    envvar="CAPELLA2POLARION_GROUPED_LINKS_CUSTOM_FIELDS",
+    is_flag=True,
+    default=True,
+)
+@click.option(
+    "--generate-figure-captions",
+    envvar="CAPELLA2POLARION_GENERATE_FIGURE_CAPTIONS",
+    is_flag=True,
+    default=False,
+)
+@click.pass_context
+def run_plugins(
+    ctx: click.core.Context,
+    plugin_config_file: typing.TextIO,
+    document_rendering_config: typing.TextIO | None,
+    overwrite_layouts: bool,
+    overwrite_numbering: bool,
+    synchronize_config: typing.TextIO | None,
+    force_update: bool,
+    type_prefix: str,
+    role_prefix: str,
+    grouped_links_custom_fields: bool,
+    generate_figure_captions: bool,
+) -> None:
+    """Synchronise model elements."""
+    capella_to_polarion_cli: Capella2PolarionCli = ctx.obj
+    logger.info(
+        "Running Plugins on Polarion project with id %s...",
+        capella_to_polarion_cli.polarion_params.project_id,
+    )
+
+    polarion_worker = pw.CapellaPolarionWorker(
+        capella_to_polarion_cli.polarion_params,
+        capella_to_polarion_cli.force_update,
+    )
+
+    configs = plugin_config.read_config_file(plugin_config_file)
+    additional_config = plugin_interfaces.AdditionalAttributes(
+        document_rendering_config,
+        overwrite_layouts,
+        overwrite_numbering,
+        synchronize_config,
+        force_update,
+        type_prefix,
+        role_prefix,
+        grouped_links_custom_fields,
+        generate_figure_captions,
+    )
+
+    for conf in configs:
+        module = importlib.import_module(str(conf.module))
+        plugin_cls: type[plugin_interfaces.PluginInterface] = getattr(
+            module, conf.plugin_name
+        )
```

Comment:
Very unsafe to load arbitrary modules provided from outside. If we want to load plugins like this we need to mention this in the documentation because this opens the door to injections. Why don't we use Python's entry-points system and force users to either hack the plugin loading or publish plugins such that we can review them and register these in the `pyproject.toml` then?

For reading:
https://packaging.python.org/en/latest/specifications/entry-points/ and https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/#using-package-metadata

Reply:
As discussed, we will present 2 approaches to the team and take a decision how to move forward. Approaches:
- Keep CLI command to execute plugins and keep using importlib
- Drop CLI command and let users implement their custom plugins as scripts and run them manually in their custom CI configs